### PR TITLE
Feat/universal link

### PR DESCRIPTION
**Pull request by Crash--**
Add support of Universal link for cozy-auth.

### CODE REVIEW

File: packages/cozy-authentication/src/utils/deeplink.js [hunk 1]
```diff
@@ -12,8 +12,29 @@ const getDeeplink = () => {
   return window.deeplinkURL
 }
 
+const generateRoute = (url, appInfos) => {
+  const { protocol, universalLinkDomain, appSlug } = appInfos
+  const urlObj = new URL(url)
+
+  //remove fallback params not needed anymore at this stage
```
Comment (ptbrowne) [inline packages/cozy-authentication/src/utils/deeplink.js]:
```suggestion
  // Remove fallback params not needed anymore at this stage
```

File: packages/cozy-authentication/src/MobileRouter.jsx [hunk 2]
```diff
@@ -69,15 +72,32 @@ export class MobileRouter extends Component {
...
   }
 
   update() {
     this.forceUpdate()
   }
+  startHandlingUniversalLinks(eventData) {
+    /* 
+   openUniversalLink seems to be called only on iOS.
+   android uses handleOpenURL by default ?!
+   */
```
Comment (ptbrowne) [inline packages/cozy-authentication/src/MobileRouter.jsx]: I find this comment not very clear. Is it to explain something ? Is it a TODO ? Please clarify

File: packages/cozy-authentication/src/utils/deeplink.js [hunk 3]
```diff
@@ -12,8 +12,29 @@ const getDeeplink = () => {
...
+    urlObj.searchParams.delete('fallback')
+  }
+  let newUrl = urlObj.toString()
+  let path = ''
+  if (newUrl.startsWith('http')) {
+    path = newUrl.replace(universalLinkDomain + '/' + appSlug, '')
+  } else {
+    path = newUrl.replace(protocol, '')
+  }
+  if (path.startsWith('/')) path = path.substring(1)
```
Comment (ptbrowne) [inline packages/cozy-authentication/src/utils/deeplink.js]: I think you can work with the urlObj a bit more without resorting to string replace that are a bit brittle. You can for example modify the protocol directly on the urlObj object `urlObj.protocol = ''` .

File: packages/cozy-authentication/src/MobileRouter.jsx (hunk 2 above)
Comment (Crash--) [reply packages/cozy-authentication/src/MobileRouter.jsx]: it's a todo. I need to double check the behavior on Android.

**Pull request merged**